ext/web/08_text_encoding.js

```diff
@@ -91,40 +92,30 @@
           allowShared: true,
         });
       }
-      options = webidl.converters.TextDecodeOptions(options, {
-        prefix,
-        context: "Argument 2",
-      });
+      let stream = false;
+      if (options !== undefined) {
+        options = webidl.converters.TextDecodeOptions(options, {
+          prefix,
+          context: "Argument 2",
+        });
+        stream = options.stream;
+      }
```

Comment:
We should come up with a general improvement in webidl for cases like this. I think the optimal way would require `new Function` codegen.

Reply:
Fwiw, Evan (Discord) had been saying that for over a year. The entire WebIDL bindinga setup should be via JiT-compiled codegen.

Reply:
yeah :(

never too late though